src/Features/CSharp/Portable/UseExplicitTypeForConst/UseExplicitTypeForConstCodeFixProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseExplicitTypeForConst
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseExplicitTypeForConst), Shared]
+    internal sealed class UseExplicitTypeForConstCodeFixProvider : CodeFixProvider
+    {
+        private const string CS0822 = nameof(CS0822); // Implicitly-typed variables cannot be constant
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(CS0822);
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            // This code fix addresses a very specific compiler error. It's unlikely there will be more than 1 of them at a time.
+            return null;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (root.FindNode(context.Span) is VariableDeclarationSyntax variableDeclaration &&
+                variableDeclaration.Variables.Count == 1)
+            {
+                var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+                var type = semanticModel.GetTypeInfo(variableDeclaration.Type, context.CancellationToken).ConvertedType;
+                if (type == null || type.TypeKind == TypeKind.Error || type.IsAnonymousType)
```

Comment:
nit: even of these, i think it would be fine to offer 'object'.  It's up to you though.

Reply:
You said you didn't like that in the previous PR (where I did the IsAnonymousType check to change that to object).

Reply:
i didn't like it in the context of the *original* "use explicit type" fixer :)   It didn't make sense there to me.  Here, it's fine since this is about an error case to begin with :)